src/java/fr/paris/lutece/portal/service/securityheader/SecurityHeaderService.java

```diff
@@ -0,0 +1,300 @@
+    	}
+    	
+		Collections.sort( securityHeadersList, Comparator.comparing( SecurityHeader::getType )
+				                                         .thenComparing( SecurityHeader::getPageCategory, Comparator.nullsLast( Comparator.naturalOrder( ) ) )
+				                                         .thenComparing( SecurityHeader::getName ) );
+    	
+		return securityHeadersList;
+    }
+	
+	/**
+	 * Returns the reference list of security headers types
+	 * 
+	 * @param locale
+	 *           The locale
+	 * @return ReferenceList object with security headers types
+	 */
+	public ReferenceList getTypeList( Locale locale )
+	{
+		ReferenceList listTypes = new ReferenceList( );
+		
+		listTypes.addItem( SecurityHeaderType.PAGE.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_TYPE, locale ) );
+		listTypes.addItem( SecurityHeaderType.REST_API.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_REST_API_TYPE, locale ) );
+		
+		return listTypes;
+	}
+	
+	/**
+	 * Returns the reference list of security headers page categories
+	 * 
+	 * @param locale
+	 *           The locale
+	 * @return ReferenceList object with security headers page categories
+	 */
+	public ReferenceList getPageCategoryList( Locale locale )
+	{
+		ReferenceList listPageCategory = new ReferenceList( );
+		
+		listPageCategory.addItem( SecurityHeaderPageCategory.ALL.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ALL, locale ) );		
+		listPageCategory.addItem( SecurityHeaderPageCategory.AUTHENTICATED_ADMIN_BACK_OFFICE.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ADMIN_BO, locale ) );
+		listPageCategory.addItem( SecurityHeaderPageCategory.AUTHENTICATED_ADMIN_FRONT_OFFICE.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ADMIN_FO, locale ) );
+		listPageCategory.addItem( SecurityHeaderPageCategory.LOGOUT_BO.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_LOGOUT_BO, locale ) );
+		listPageCategory.addItem( SecurityHeaderPageCategory.LOGOUT_FO.getCode( ), I18nService.getLocalizedString( PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_LOGOUT_FO, locale ) );
+		
+		return listPageCategory;
+	}
+	
+	/**
+	 * Create the security header given in parameter. Like all the actions, this one invalidates the map containing the security headers.
+	 * 
+	 * @param securityHeader
+	 *                The security header to create
+	 */
+	public void create( SecurityHeader securityHeader )
+	{
+		_logger.debug( "Security header to create : name : {}, value : {}, type : {}, page category : {}", securityHeader.getName( ), securityHeader.getValue( ), securityHeader.getType( ), securityHeader.getPageCategory( ) );
+		SecurityHeaderHome.create( securityHeader );
+		clearMapSecurityHeaders( );
```

Comment:
The management of the refresh of the cache of the list of active headers could be as follows:

In the JSPbean managing these security headers:
- we can only modify a header if it is inactive
- when we create a header, it is inactive by default
- each time we activate or deactivate a header, we call the SecurityHeaderService.refresh() method (which will automatically refresh the cache of the map of active headers)

Reply:
point 1 taken into account in commit 93a56b8139c008a7cc3a87cce5a8ccbffb4c9734. Initial development was already taking into account points 2 and 3